src/moin/storage/middleware/protecting.py

```diff
@@ -162,36 +177,65 @@ def search_meta(self, q, idx_name=LATEST_REVS, **kw):
         The intended use of this method is to return the current rev metadata for all
         of the items in namespace subject to query restrictions. This is useful for reports
         such as Global Index, Global Tags, Wanted Items, Orphaned Items, etc.
-
-        Save processing time by avoiding a full ACL check when the answer will be the same as the last.
         """
+        from . indexing import parent_names
```

Comment:
does that work? usually one writes `from .indexing ...`.

Reply:
will fix